### ISSUE TITLE
Can/should Hasura handle rollbacks when the Github deployment feature errors out

### ISSUE THREAD

**Issue by dsandip** [hasura/graphql-engine#9067]
### Is your proposal related to a problem?

E.g. use-case: when applying migrations fails, the metadata is likely to be wrong/inconsistent. 

### Describe the solution you'd like
There are clear instructions on what to do when an issue happens (redirect to [docs troubleshooting section](https://hasura.io/docs/latest/deployment/hasura-cloud/github-integration/#inconsistent-metadata-after-finalizing-update) : `rollback can be done to the previous functioning state of metadata by reverting your metadata changes on your branch and then pushing to redeploy.`). However, can this integration also handle this rollback or document why it can't do so?

**Comment by chrysb** [comment]
Yes please, rolling back the metadata to the previous functioning state upon failed migration would be essential.

**Comment by thecoolestcoderrushil** [comment]
Failed migrations leaving metadata in an inconsistent state is exactly why rollback matters, and without it every failed deploy needs manual cleanup. Rehearsing migrations on a production-shaped twin first reduces how often you need that rollback at all. antifailure.dev (https://antifailure.dev) spins up disposable Postgres twins for migration rehearsal, and it is open source and free to try. Happy to set up a demo if that would help.
